src/Tempest/Generation/src/HasGeneratorCommand.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Generation;
+
+use Tempest\Console\Console;
+use Tempest\Core\Composer;
+use Tempest\Support\PathHelper;
+use function Tempest\Support\str;
+use Tempest\Validation\Rules\EndsWith;
+use Tempest\Validation\Rules\NotEmpty;
+
+/**
+ * This provide bunch of methods to generate files.
+ * Also methods to manipulate the user generic input like the class name.
+ */
+trait HasGeneratorCommand
+{
+    public function __construct(
+        protected readonly Console $console,
+        protected readonly Composer $composer,
+        protected readonly StubFileGenerator $stubFileGenerator
+    ) {
+    }
+
+    /**
+     * Get a suggested path for the given class name.
+     * This will find the main autoloaded namespace in the project's composer file and use it as the base path.
+     *
+     * @param string $className The class name to generate the path for, can include path parts (e.g. 'Models/User').
+     * @param string|null $pathPrefix The prefix to add to the path (e.g. 'Models').
+     * @param string|null $classSuffix The suffix to add to the class name (e.g. 'Model').
+     *
+     * @return string The fully suggested path including the filename and extension.
```

Comment:
```suggestion
     * Gets a suggested path for the given class name.
     * This will use the user's main namespace as the base path.
     *
     * @param string $className The class name to generate the path for, can include path parts (e.g. 'Models/User').
     * @param string|null $pathPrefix The prefix to add to the path (e.g. 'Models').
     * @param string|null $classSuffix The suffix to add to the class name (e.g. 'Model').
     *
     * @return string The fully suggested path including the filename and extension.
```